packages/utils/upgrade/README.md

```diff
@@ -1 +1,179 @@
+    // Return the json object unmodified to pass it to the next transform
+    return file.json;
+  }
+
+  // Use json() to get useful helpers for performing your transform
+  const j = json(file.json);
+
+  const strapiDepAddress = 'dependencies.@strapi/strapi';
+
+  // if this file contains a value at dependencies.@strapi/strapi
+  if (j.has(strapiDepAddress)) {
+    // we set the value to 5.0.0
+    j.set(strapiDepAddress, '5.0.0');
+  }
+
+  // at the end we must return the modified json object
+  return j.root();
+};
+
+export default transform;
+```
+
+For reference, these are the types for the relevant objects, which can be found in `packages/utils/upgrade/src/core/runner/json.ts`:
+
+```typescript
+export interface JSONTransformParams {
+  cwd: string;
+  json: (object: Utils.JSONObject) => JSONTransformAPI;
+}
+
+export interface JSONTransformAPI {
+  get<T extends Utils.JSONValue>(path?: string, defaultValue?: T): T | undefined;
+  has(path: string): boolean;
+  set(path: string, value: Utils.JSONValue): this;
+  merge(other: Utils.JSONObject): this;
+  root(): Utils.JSONObject;
+}
+
+export type JSONTransform = (file: JSONSourceFile, params: JSONTransformParams) => Utils.JSONObject;
+```
+
+The methods available from `json()` are wrappers for the lodash methods of the same name:
+
+- **get(path, default)**: get path or default if not found
+- **set(path, value)**: set path (such as 'engines.node', 'dependencies', 'author.name') to value
+- **has(path)**: checks if path exists
+- **merge(obj)**: merges two json objects
+- **root()**: returns the whole json object
+
+### 'code' codemod transforms
+
+Codemod transforms use the [`jscodeshift`](https://github.com/facebook/jscodeshift) library to modify code passed in. Please see their documentation for advanced details.
+
+```typescript
+import type { Transform } from 'jscodeshift';
+
+const transform: Transform = (file, api) => {
+  // Extract the jscodeshift API
+  const { j } = api;
+  // Parse the file content
+  const root = j(file.source);
+
+  root
+    // Find console.log calls expressions
+    .find(j.CallExpression, {
+      callee: { object: { name: 'console' }, property: { name: 'log' } },
+    })
+    // For each call expression
+    .forEach((path) => {
+      const { callee } = path.node;
+
+      if (
+        // Make sure the callee is a member expression (object/property)
+        j.MemberExpression.check(callee) &&
+        // Make sure the property is an actual identifier (contains a name property)
+        j.Identifier.check(callee.property)
+      ) {
+        // Update the property's identifier name
+        callee.property.name = 'info';
+      }
+    });
+
+  // Return the updated file content
+  return root.toSource();
+};
+
+export default transform;
+```
+
+For reference, these are the types for the relevant objects, which can be found in `packages/utils/upgrade/src/core/runner/code.ts`:
+
+```typescript
+export interface CodeRunnerConfig {
```

Comment:
I don't think it makes sense to share the types here, since they're not used in the transforms directly